builder/amazon/common/step_modify_ebs_instance.go

```diff
@@ -37,20 +38,24 @@ func (s *StepModifyEBSBackedInstance) Run(_ context.Context, state multistep.Sta
 		}
 	}
 
-	// Set EnaSupport to true.
+	// Handle EnaSupport flag.
 	// As of February 2017, this applies to C5, I3, P2, R4, X1, and m4.16xlarge
+	var prefix string
 	if s.EnableAMIENASupport {
-		ui.Say("Enabling Enhanced Networking (ENA)...")
-		_, err := ec2conn.ModifyInstanceAttribute(&ec2.ModifyInstanceAttributeInput{
-			InstanceId: instance.InstanceId,
-			EnaSupport: &ec2.AttributeBooleanValue{Value: aws.Bool(true)},
-		})
-		if err != nil {
-			err := fmt.Errorf("Error enabling Enhanced Networking (ENA) on %s: %s", *instance.InstanceId, err)
-			state.Put("error", err)
-			ui.Error(err.Error())
-			return multistep.ActionHalt
-		}
+		prefix = "En"
+	} else {
+		prefix = "Dis"
+	}
```

Comment:
I was just thinking, what if `s.EnableAMIENASupport` was a pointer; then if it's `nil` then we keep whatever setting we had.

Reply:
That would make sense as it seemed to me that the setting was either `true` or `false`, there was no undefined state. Would that be preferred?

Reply:
My understanding is that this PR's change forces true vs false rather than having an undefined state, which technically changes the behavior; prior to this change the options were essentially "true" or "undefined", or you wouldn't have had the issues you saw.  Using @azr's suggestion would allow us to maintain backwards behavior where not setting the value means that we won't change it either way.

Reply:
Ok, I will see if I can rework this to be a pointer